docs/spec/v1beta1/kustomization.md

```diff
@@ -564,6 +584,65 @@ spec:
       newTag: 5.0.0
 ```
 
+## Variable substitution
+
+With `spec.postBuild.substitute` you can provide a map of key/value pairs holding the
+variables to be substituted in the final YAML manifest, after kustomize build.
+
+This offers basic templating for your manifests including support
+for [bash string replacement functions](https://github.com/drone/envsubst) e.g.:
+
+- `${var:=default}`
+- `${var:position}`
+- `${var:position:length}`
+- `${var/substring/replacement}`
+
+Assuming you have manifests with the following variables:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: apps
+  labels:
+    environment: ${env:=dev}
+    region: "${region}"
+```
+
+You can specify the variables and their values in the Kustomization definition under
+the `substitute` post build section:
+
+````yaml
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
+kind: Kustomization
+metadata:
+  name: apps
+spec:
+  interval: 5m
+  path: "./apps/"
+  postBuild:
+    substitute:
+      env: "prod"
+      region: "eu-central-1"
+````
+
+You can replicate the controller post-build substitutions locally using
+[kustomize](https://github.com/kubernetes-sigs/kustomize)
+and Drone's [envsubst](https://github.com/drone/envsubst):
+
+```console
+$ export region=eu-central-1
+$ kustomize build ./apps/ | envsubst 
```

Comment:
`envsubst` on most systems will not be Drone's Go package, but [the `envsubst` helper from GNU gettext](https://linux.die.net/man/1/envsubst). I would thus not name the Drone dependency here.

Reply:
The GNU envsubst doesn't support bash functions and thus can't be used to replicate the controller functionality locally.

Reply:
In that case I would highlight the difference between `envsubst` from Drone and the version that most people will have installed on their machine.